proxy/proxy.go

```diff
@@ -866,12 +868,17 @@ func (p *Proxy) makeUpgradeRequest(ctx *context, req *http.Request) error {
 
 func (p *Proxy) makeBackendRequest(ctx *context) (*http.Response, *proxyError) {
 	var err error
-	req, err := mapRequest(ctx.request, ctx.route, ctx.outgoingHost, p.flags.HopHeadersRemoval(), ctx.StateBag())
+	req, endpoint, err := mapRequest(ctx.request, ctx.route, ctx.outgoingHost, p.flags.HopHeadersRemoval(), ctx.StateBag())
 	if err != nil {
 		p.log.Errorf("could not map backend request, caused by: %v", err)
 		return nil, &proxyError{err: err}
 	}
 
+	if endpoint != nil {
+		endpoint.Metrics.IncInflightRequest()
```

Comment:
What do you think about changing it to something more generic?
E.g. with
```
start := endpoint.Metrics.RequestStarted()
defer endpoint.Metrics.RequestFinished(start)
```
it may count inflight requests as well as measure latency (tbd).

Reply:
`metrics.Metrics.MeasureSince` can be used.

currently we use a timer for the backend host, but it doesn't consider the endpoints. Another problem of it is that it includes more than just what it strictly needs to, e.g. the retries with a different endpoint.

Reply:
sorry, revisiting my comment about MeasureSince here, because it is a 'heavier' tool. The current atomic counter is a good start.